core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/command/handlers/CancelTransferCommandHandler.java

```diff
@@ -30,7 +31,14 @@ public CancelTransferCommandHandler(TransferProcessStore store) {
 
     @Override
     protected boolean modify(TransferProcess process) {
-        process.transitionError("Cancelled");
+        var state = process.getState();
+        if (state == TransferProcessStates.COMPLETED.code() ||
```

Comment:
a `canBeCancelled` or `isInFinalState` method on `TransferProcess` could make this more readable and better encapsulate the knowledge.

Reply:
I'm hesitant to add that method for just that one usage. If we ever need to check the same condition again, I'm all for encapsulating it though.

Reply:
As a rule of thumb, the less the states are valued outside the `TransferProcess`, the better. This is a way to stay away from an anemic model.
But ok, I agree on waiting for duplication.